### PR TITLE
ENH: Add a 3D renderer to the pose solver GUI, will show poses

### PR DESCRIPTION
**Pull request by vaughantnrc**
Renderer is shown in screenshot.

image

### CODE REVIEW

**Comment by vaughantnrc** [comment]
#25

**Comment by Rampex1** [comment]
Please add install instructions for VS Build Tools

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,12 @@
 fastapi==0.100.0
 fastapi-utils==0.2.1
 hjson
+numpy
+numpy-stl
 opencv-contrib-python==4.5.5.64
 pydantic~=1.10.15
+PyOpenGL==3.1.7
+PyOpenGL-accelerate==3.1.7
```
Comment (Rampex1) [inline requirements.txt:9]: Add instructions to download visual studio build tools.
Specify compatible Pycharm versions. (2023)

Comment (vaughantnrc) [reply requirements.txt:9]: I added instructions to `doc/setup.md` about Build Tools.

I'm not sure that there's an incompatibility between this code and a specific version of PyCharm. It is possible that we saw a bug in PyCharm, or that there is an incompatibility issue with our own computer configurations at the NRC. So the cause is unknown...

**Comment by lassoan** [comment]
We should not write any new code related to OpenGL: 1. It is unnecessarily low level (we don't need optimizations that are only available at such low level), 2. OpenGL support is being phased out. Using VTK would be an obvious choice for many reasons, but if for some reason we must use a low-level API then we should use WebGPU (it is not only for web or JavaScript, but it is the new universal GPU API).

**Comment by vaughantnrc** [comment]
Thank you Andras. It is good to be aware of these alternatives. I don't know that there is currently a universal GPU API. I think vtk would be preferable (as a layer over whichever GPU API implementation) if we were able to get it to run within the GUI (currently based on wxPython). I certainly think it's worth noting as a future issue.

**Pull request merged**